src/types/mod.rs

```diff
@@ -0,0 +1,54 @@
+mod errors;
+use self::errors::SerializableError;
+use alloy::{
+    primitives::{B256, U256},
+    rpc::types::eth::TransactionReceipt,
+};
+use serde::{Deserialize, Serialize};
+pub trait Serializable {
+    fn to_bin(&self) -> Result<Vec<u8>, Box<bincode::ErrorKind>>;
+    fn from_bin(data: Vec<u8>) -> Result<Self, SerializableError>
+    where
+        Self: Sized;
+}
+
+/// Describes the structure of a payment method in
+/// a gateway
+#[derive(Clone, Deserialize, Serialize)]
+pub struct PaymentMethod {
+    /// Whether or not the method uses native gas token
+    pub is_native: bool,
+    /// The address of the ERC20 token
+    pub token_address: Option<String>,
+}
+
+#[derive(Clone, Deserialize, Serialize)]
+pub struct Invoice {
+    /// Recipient address
+    pub to: String,
+    /// Recipient instance
+    pub wallet: B256,
```

Comment:
use https://docs.rs/zeroize/latest/zeroize/
it will make it zeroes after drop.
in necessary because if anyone sends more value than expected in case of security issue it can lead to the loosing of funds

Reply:
e889da77a756abf1d3390a98806640e31f58cfb7